ui/core.js

```diff
@@ -126,6 +128,67 @@ $.fn.extend( {
 				$( this ).removeAttr( "id" );
 			}
 		} );
+	},
+
+	form: function() {
+		var form;
+
+		if ( this[ 0 ].form && typeof this[ 0 ].form !== "string" ) {
+			return this.pushStack( $( this[ 0 ].form ) );
+		} else if ( !this[ 0 ].form ) {
+			return this.pushStack( [] );
+		}
+
+		// Support: IE8 only ( the rest of the method )
+		// IE8 supports the form property, but not the form attribute, worse yet if you supply the
+		// form attribute it overwrites the form property with the string. Other supported browsers
+		// like all other IE's and Android 2.3 support the form attribute not at all or partially.
+		// We don't care in those cases because they still always return an element. We are not
+		// trying to fix the form attribute here, only deal with the prop supplying a string.
+		form = this.closest( "form" );
+		if ( form.length ) {
+			return this.pushStack( form );
+		}
+		return this.pushStack( [] );
+	},
+
+	labels: function() {
+		var ancestor, selector, id, labels, ancestors;
+
+			// Check control.labels first
+			if ( this[ 0 ].labels !== undefined && this[ 0 ].labels.length > 0 ) {
+				return this.pushStack( this[ 0 ].labels );
+			}
+
+			// Support: IE <= 11, FF <= 37, Android <=2.3 only
+			// Above browsers do not support control.labels everything below is to support them
+			// as well as document fragments control.labels does not work on document fragments anywhere
+			labels = this.parents( "label" );
```

Comment:
I debated this. The reason i didn't was the early return here does not gain us anything, there is still only one conditional no logic is skipped,  it only really serves to add an extra return. I also thought the intended logic was clearer this way since the idea is to return all the labels never just a subset. 

Im fine with changing if you prefer the early return since this is what we generally prefer but that was my reasoning being the current  code organization.